OpenRA.Game/Actor.cs

```diff
@@ -279,30 +279,38 @@ public void Dispose()
 			});
 		}
 
-		// TODO: move elsewhere.
-		public void ChangeOwner(Player newOwner)
+		private void ChangeOwnerInternal(Player newOwner, World world)
 		{
-			World.AddFrameEndTask(w =>
-			{
-				if (Disposed)
-					return;
+			if (Disposed)
+				return;
 
-				var oldOwner = Owner;
-				var wasInWorld = IsInWorld;
+			var oldOwner = Owner;
+			var wasInWorld = IsInWorld;
 
-				// momentarily remove from world so the ownership queries don't get confused
-				if (wasInWorld)
-					w.Remove(this);
+			// momentarily remove from world so the ownership queries don't get confused
+			if (wasInWorld)
+				world.Remove(this);
 
-				Owner = newOwner;
-				Generation++;
+			Owner = newOwner;
+			Generation++;
 
-				foreach (var t in TraitsImplementing<INotifyOwnerChanged>())
-					t.OnOwnerChanged(this, oldOwner, newOwner);
+			foreach (var t in TraitsImplementing<INotifyOwnerChanged>())
+				t.OnOwnerChanged(this, oldOwner, newOwner);
 
-				if (wasInWorld)
-					w.Add(this);
-			});
+			if (wasInWorld)
+				world.Add(this);
+		}
+
+		// TODO: move elsewhere.
+		public void ChangeOwner(Player newOwner, World externalWorld = null)
+		{
+			if (externalWorld == null)
+				World.AddFrameEndTask(w =>
```

Comment:
You should add braces to the `if` here as the body spans over multiple lines, imho.